hyperspy/_components/polynomial.py

```diff
@@ -64,8 +64,11 @@ def grad_coefficients(self, x):
             return x * 0
 
     def __repr__(self):
-        return u'Component <%s order polynomial>'.encode('utf8') % (
-            ordinal(self.get_polynomial_order()))
+        text = u"%s order Polynomial component" % ordinal(
+            self.get_polynomial_order())
+        if self.name:
+            text = u"%s (%s)" % (self.name, text)
+        return u"<%s>".encode('utf8') % text
```

Comment:
Huh, good point - it will never have to deal with any strings from "outside", hence should be fine. I removed it